File: convalida-validators/src/test/java/convalida/validators/ConfirmEmailValidatorTest.java
package convalida.validators;

import android.text.Editable;
import android.widget.EditText;

import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

/**
 * @author Wellington Costa on 02/11/2017.
 */
public class ConfirmEmailValidatorTest extends BaseTest {

    private EditText emailEditText;

    @Before public void setUpValidators() {
        emailEditText = mock(EditText.class);
        when(emailEditText.getText()).thenReturn(mock(Editable.class));
    }

    @Test public void emptyValues() {
        ConfirmEmailValidator validator = new ConfirmEmailValidator(emailEditText, mockEditText, errorMessage, true);
        when(emailEditText.getText().toString()).thenReturn("");
        when(mockEditText.getText().toString()).thenReturn("");
        assertEquals(validator.validate(), true);
    }

    @Test public void emailsDoesNotMatch() {
        ConfirmEmailValidator validator = new ConfirmEmailValidator(emailEditText, mockEditText, errorMessage, true);
        when(emailEditText.getText().toString()).thenReturn("wellington@email.com");
        when(mockEditText.getText().toString()).thenReturn("wellington@email.co");
        assertEquals(validator.validate(), false);
    }

    @Test public void emailsMatch() {
        ConfirmPasswordValidator validator = new ConfirmPasswordValidator(emailEditText, mockEditText, errorMessage, true);
        when(emailEditText.getText().toString()).thenReturn("wellington@email.com");
        when(mockEditText.getText().toString()).thenReturn("wellington@email.com");
        assertEquals(validator.validate(), true);
    }

}

File: convalida-validators/src/test/java/convalida/validators/ConfirmEmailValidatorTest.kt
package convalida.validators

import android.text.Editable
import android.widget.EditText
import org.junit.Assert.assertEquals
import org.junit.Before
import org.junit.Test
import org.mockito.Mockito.`when`
import org.mockito.Mockito.mock

/**
 * @author Wellington Costa on 02/11/2017.
 */
class ConfirmEmailValidatorTest : BaseTest() {

    private lateinit var emailEditText: EditText

    @Before
    fun setUpValidators() {
        emailEditText = mock(EditText::class.java)
        `when`(emailEditText.text).thenReturn(mock(Editable::class.java))
    }

    @Test
    fun emailsDoesNotMatch() {
        val validator = ConfirmEmailValidator(emailEditText, mockEditText, errorMessage, true)
        `when`(emailEditText.text.toString()).thenReturn("wellington@email.com")
        `when`(mockEditText.text.toString()).thenReturn("wellington@email.co")
        assertEquals(validator.validate(), false)
    }

    @Test
    fun emailsMatch() {
        val validator = ConfirmPasswordValidator(emailEditText, mockEditText, errorMessage, true)
        `when`(emailEditText.text.toString()).thenReturn("wellington@email.com")
        `when`(mockEditText.text.toString()).thenReturn("wellington@email.com")
        assertEquals(validator.validate(), true)

    }

}